### ISSUE TITLE
oneflow.softmax perform differently between cpu and cuda.

### ISSUE THREAD

**Issue by Redmept1on** [Oneflow-Inc/oneflow#10469]
## Summary

oneflow.softmax perform differently between cpu and cuda.

## Code to reproduce bug

```python
import oneflow as flow
import numpy as np

x1 = flow.tensor(np.array([[float('inf'), 0, -1, float('nan'), 5]], dtype=np.float32))
x1 = x1.cuda()
y1 = flow.softmax(x1)
print(y1.device,y1)

x1 = flow.tensor(np.array([[float('inf'), 0, -1, float('nan'), 5]], dtype=np.float32))
x1 = x1.cpu()
y2 = flow.softmax(x1)
print(y2.device,y2)

```
![image](https://github.com/Oneflow-Inc/oneflow/assets/84373897/461af52c-090c-4835-994d-567879870d70)

pytorch
```python
import torch
import numpy as np

input_tensor = torch.tensor(np.array([[float('inf'), 0, -1, float('nan'), 5]], dtype=np.float32))
# other_tensor = torch.tensor(np.array([[float('nan'), 0, -1, float('nan'), 5]], dtype=np.float32))
output_tensors = torch.nn.functional.softmax(input_tensor)

print(output_tensors)
```
![image](https://github.com/Oneflow-Inc/oneflow/assets/84373897/05dc946b-6acd-465e-9517-ccad2c5319de)

## System Information

- What is your OneFlow installation :dockerhub
- OS:Linux
- OneFlow version (run python3 -m oneflow --doctor):0.9.1.dev20231024+cu117
- Python version: 3.8
